File: codility/CyclicRotation.js
// Rotate an array to the right by a given number of steps.

function solution(A, K) {
    // write your code in JavaScript (Node.js 8.9.4)
    const shift = K % A.length;
    if (shift === 0) {
        return A;
    }
    const rem = A.length - shift;
    const result = [];
    for (let i = rem; i < A.length; i++) {
        result.push(A[i]);
    }
    for (let i = 0; i < rem; i++) {
        result.push(A[i]);
    }
    return result;
}

File: codility/cyclic_rotation.go
package solution

// you can also use imports, for example:
// import "fmt"
// import "os"

// you can write to stdout for debugging purposes, e.g.
// fmt.Println("this is a debug message")

func Solution(A []int, K int) []int {
	// write your code in Go 1.4
	size := len(A)
	if size < 2 || K%size == 0 {
		return A
	}
	result := make([]int, size)
	for i, v := range A {
		result[(i+K)%size] = v
	}
	return result
}
